### ISSUE TITLE
consistency in maven/packaging versions

### ISSUE THREAD

**Issue by landryb** [georchestra/georchestra#1302]
So far i've used packages built from master using 15.12-SNAPSHOT version, which produced debian packages with for example 15.12+201602011716 for the version. Now that 15.12 is released, i built packages from the 15.12 branch, produced packages with 15.12-1 version.. problem is, those packages are 'older' in terms of versions than the ones built before, and by default apt wont upgrade a package with 15.12+2016something to 15.12-1.

Since the package version comes from the mvn version we use in pom.xml files, i think we should fix it at the source, and instead of doing (like we did) '15.12-SNAPSHOT -> 15.12 -> 16.06-SNAPSHOT -> 16.06' where there's a *downgrade* in versions, we should be doing '15.12 -> 15.12-SNAPSHOT -> 16.06 -> 16.06-SNAPSHOT', as 16.06-SNAPSHOT is *after* 16.06 and doesnt mean 'pre-release-snapshot-of-what-will-be-16.06'.

To sum it up:
- in master, now we should have 15.12-SNAPSHOT for all versions (thus, revert 577fc80971acf6302ad153c534e91e30a2997016)
- in 15.12, we should have 15.12 in pom.xml (.something ?)
- when we release 16.06, master moves to 16.06-SNAPSHOT, and the 16.06 branch gets 16.06 as version

@fvanderbiest does this make sense to you, or it goes against maven concept of what SNAPSHOT means ? I can cook a PR to fix RELEASE_PROCEDURE.md

**Comment by landryb** [comment]
Looking a bit at maven docs, for them 15.12-SNAPSHOT means 'what will become 15.12'. I think this is totally wrong, but that's with my long-time-packager hat :)

**Comment by fvanderbiest** [comment]
We are in line with the maven docs, indeed. I think the problem lies somewhere else than the pom versions. 

Maybe we should name packages differently, eg 15.12-SNAPSHOT -> 15.06+201602011716 ?

**Comment by fvanderbiest** [comment]
Or 15.12-SNAPSHOT -> dev+201602011716 ?

**Comment by fvanderbiest** [comment]
Or maybe we should not mix dev packages and releases in the same repo ?

**Comment by fvanderbiest** [comment]
Anyway, I'm -1 on changing the way we name versions in poms.

**Comment by pmauduit** [comment]
> Anyway, I'm -1 on changing the way we name versions in poms.

I'm personally perplexed (what would be the more relevant thing to do ? Following debian packaging conventions on naming the packages, or the maven ones ?)

**Comment by landryb** [comment]
then that means we need to add a step to RELEASE_PROCEDURE.md to set packageVersion in the configuration of debian-maven-plugin in all projects - and fix it to something 'lesser' than the release one when on the master branch. And then i think we lose the +date scheme.. 

https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-Version for reference. maybe we should use ~dev.
